### PR TITLE
Improve error message for invalid path dependency

### LINKED ISSUE #23357: Improve error message for invalid path dependency
For an invalid path dependency specified in the `Ballerina.toml` file, I get an error message of the following format:
```
error: balo file for dependency [pubudu/bar] does not exists: <absolute_path_to_balo_file>
```
What's the correct way to refer to `.balo` files? The https://ballerina.io/learn/how-to-structure-ballerina-code/ guide refer to them as BALO files. I think here also we either need to use that or `.balo` form. 

Also, it should be be rephrased in a better way. How about something like,
```
.balo file for the 'pubudu/bar' module not found: <specified_path>
```

### PR DESCRIPTION
**Pull request by suganyasuven**
Fixes #23357

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pubudu91** [comment]
Did you check what the correct term to use would be for balo files? If it's balo files, can you create an issue in the website repo as well, to correct the terminology in the website?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java [hunk 1]
```diff
@@ -80,14 +80,14 @@ public Patten calculate(PackageID moduleID) {
         
         // if balo file does not exists
         if (Files.notExists(baloPath)) {
-            throw new BLangCompilerException("balo file for dependency [" + dep.getModuleID() + "] does not exists: " +
-                                             dep.getMetadata().getPath().toAbsolutePath().normalize());
+            throw new BLangCompilerException("balo file for module [" + dep.getModuleID() + "] is not found in the " +
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:83]:
```suggestion
            throw new BLangCompilerException("balo file for module [" + dep.getModuleID() + "] was not found in the " +
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java [hunk 2]
```diff
@@ -80,14 +80,14 @@ public Patten calculate(PackageID moduleID) {
...
+                    "specified path: " + dep.getMetadata().getPath());
         }
     
         // if balo file is not a file
         if (!Files.isRegularFile(baloPath)) {
-            throw new BLangCompilerException("balo file for dependency [" + dep.getModuleID() + "] is not a file: " +
-                                             dep.getMetadata().getPath().toAbsolutePath().normalize());
+            throw new BLangCompilerException("balo file for module [" + dep.getModuleID() + "] is not a regular " +
+                    "file: " + dep.getMetadata().getPath().toAbsolutePath().normalize());
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:90]: What's meant by "a regular file"? Wouldn't it be better to simply say that the provided file is not a balo file?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java (hunk 1 above)
Comment (praneesha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:83]: +1

Comment (praneesha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:83]:
```suggestion
            throw new BLangCompilerException("The BALO file of the [" + dep.getModuleID() + "] module was not found in the " +
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java (hunk 2 above)
Comment (praneesha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:90]: +1

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java [hunk 3]
```diff
@@ -80,14 +80,14 @@ public Patten calculate(PackageID moduleID) {
...
-                                             dep.getMetadata().getPath().toAbsolutePath().normalize());
+            throw new BLangCompilerException("balo file for module [" + dep.getModuleID() + "] is not found in the " +
+                    "specified path: " + dep.getMetadata().getPath());
         }
     
         // if balo file is not a file
         if (!Files.isRegularFile(baloPath)) {
-            throw new BLangCompilerException("balo file for dependency [" + dep.getModuleID() + "] is not a file: " +
-                                             dep.getMetadata().getPath().toAbsolutePath().normalize());
+            throw new BLangCompilerException("balo file for module [" + dep.getModuleID() + "] is not a regular " +
```
Comment (praneesha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/repo/PathBaloRepo.java:89]:
```suggestion
            throw new BLangCompilerException("The BALO file of the [" + dep.getModuleID() + "] module is not a regular " +
```

**Comment by hevayo** [comment]
Shall we change the file extension to ".balo" please refer https://english.stackexchange.com/questions/291003/usage-of-file-extensions-as-nouns-and-verbs it seems industry best practices use .ext in simple case for file extensions.

**Comment by praneesha** [comment]
> Shall we change the file extension to ".balo" please refer https://english.stackexchange.com/questions/291003/usage-of-file-extensions-as-nouns-and-verbs it seems industry best practices use .ext in simple case for file extensions.
> 
> Also we can adopt this in the website as well.

+1

**Comment by pubudu91** [comment]
@praneesha Can you look in to that? The https://ballerina.io/learn/how-to-structure-ballerina-code/ page in particular has a lot of references to .balo files as "BALO files".

**Pull request closed without merging**